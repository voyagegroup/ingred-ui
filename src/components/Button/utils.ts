import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
