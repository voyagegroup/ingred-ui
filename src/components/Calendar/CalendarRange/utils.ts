import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
