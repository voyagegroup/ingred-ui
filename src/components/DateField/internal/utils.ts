export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
