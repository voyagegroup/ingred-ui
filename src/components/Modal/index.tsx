export { default } from "./Modal";
export type { ModalProps, ModalCloseReason } from "./Modal";
