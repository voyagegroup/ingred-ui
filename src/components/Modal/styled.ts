import styled from "styled-components";

export const Container = styled.div<{ isHidden: boolean }>`
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  z-index: ${({ theme }) => theme.depth.modal};

  visibility: ${({ isHidden }) => (isHidden ? "hidden" : "visible")};
`;
