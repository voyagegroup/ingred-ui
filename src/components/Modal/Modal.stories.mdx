import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import Modal from "./Modal";
import Button from "../Button";
import Fade from "../Fade";

<Meta
  title="Components/Utils/Modal"
  component={Modal}
  parameters={{ docs: { source: { type: "code" } } }}
/>

# Modal

<ArgsTable of={Modal} />

## Samples

<Canvas>
  <Story
    name="Example"
    args={{
      isOpen: false,
    }}
  >
    {(args) => {
      const [isOpen, setIsOpen] = React.useState(args.isOpen);
      const handleToggleOpen = () => {
        setIsOpen(!isOpen);
      };
      return (
        <>
          <Button onClick={handleToggleOpen}>Toggle Show Modal</Button>
          <Modal {...args} isOpen={isOpen} onClose={handleToggleOpen}>
            <Fade in={isOpen}>
              <div
                style={{
                  position: "absolute",
                  top: "50%",
                  left: "50%",
                  width: "50vw",
                  height: "50vh",
                  transform: "translate(-50%, -50%)",
                  backgroundColor: "white",
                  display: "flex",
                  flexDirection: "column",
                  justifyContent: "center",
                  alignItems: "center",
                }}
              >
                <div>This is Modal.</div>
                <Button inline={true} onClick={handleToggleOpen}>
                  Close
                </Button>
              </div>
            </Fade>
          </Modal>
        </>
      );
    }}
  </Story>
</Canvas>
