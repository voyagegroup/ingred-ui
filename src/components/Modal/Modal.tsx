import * as React from "react";
import * as Styled from "./styled";
import Portal from "../Portal";
import Backdrop, { BackdropProps } from "../Backdrop";
import { createChainedFunction } from "../../utils/createChainedFunction";

const getHasTransition = (props: React.PropsWithChildren<any>): boolean => {
  // eslint-disable-next-line no-prototype-builtins
  return props.hasOwnProperty("in");
};

// TODO: enable close with Escape key
export type ModalCloseReason = "backdropClick";

export type ModalProps = {
  /**
   * If `false`, children becomes `visibility: hidden`.
   */
  isOpen?: boolean;
  /**
   * If `true`, children contains [Backdrop](/?path=/docs/components-feedback-backdrop).
   */
  hasBackground?: boolean;
  /**
   * props of [Backdrop](/?path=/docs/components-feedback-backdrop)
   */
  backdropProps?: BackdropProps;
  /**
   * Control the transition of Backdrop.
   */
  enableTransition?: boolean;
  onClose?: (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
    reason: ModalCloseReason,
  ) => void;
  children: React.ReactElement;
};

const Modal = React.forwardRef<HTMLDivElement, ModalProps>(function MenuList(
  {
    isOpen = true,
    hasBackground = true,
    backdropProps,
    enableTransition = true,
    onClose,
    children,
  },
  ref,
) {
  const [exited, setExited] = React.useState<boolean>(true);

  const handleBackDropClick = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    if (onClose) onClose(event, "backdropClick");
  };

  const handleEnter = () => {
    setExited(false);
  };

  const handleExited = () => {
    setExited(true);
  };

  const childProps: any = {};
  if (enableTransition && getHasTransition(children.props)) {
    childProps.onEnter = createChainedFunction(
      handleEnter,
      children.props.onEnter,
    );
    childProps.onExited = createChainedFunction(
      handleExited,
      children.props.onExited,
    );
  }

  return (
    <Portal>
      <Styled.Container ref={ref} isHidden={!isOpen && exited}>
        {hasBackground && (
          <Backdrop
            {...backdropProps}
            isOpen={!enableTransition || isOpen}
            onClick={handleBackDropClick}
          />
        )}
        {React.cloneElement(children, childProps)}
      </Styled.Container>
    </Portal>
  );
});

export default Modal;
