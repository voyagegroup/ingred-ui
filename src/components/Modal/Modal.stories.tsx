import React from "react";
import { Story } from "@storybook/react/types-6-0";
import { Title, ArgsTable, Stories } from "@storybook/addon-docs";
import Modal, { ModalProps } from "./Modal";
import Button from "../Button";
import Fade from "../Fade";

export default {
  title: "Components/Utils/Modal",
  component: Modal,
  args: {
    isOpen: false,
  },
  parameters: {
    docs: {
      source: { type: "code" },
      page: () => (
        <>
          <Title />
          <ArgsTable of={Modal} />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

export const Example: Story<ModalProps> = (args) => {
  const [isOpen, setIsOpen] = React.useState(args.isOpen);
  const handleToggleOpen = () => {
    setIsOpen(!isOpen);
  };
  return (
    <>
      <Button onClick={handleToggleOpen}>Toggle Show Modal</Button>
      <Modal {...args} isOpen={isOpen} onClose={handleToggleOpen}>
        <Fade in={isOpen}>
          <div
            style={{
              position: "absolute",
              top: "50%",
              left: "50%",
              width: "50vw",
              height: "50vh",
              transform: "translate(-50%, -50%)",
              backgroundColor: "white",
              display: "flex",
              flexDirection: "column",
              justifyContent: "center",
              alignItems: "center",
            }}
          >
            <div>This is Modal.</div>
            <Button inline={true} onClick={handleToggleOpen}>
              Close
            </Button>
          </div>
        </Fade>
      </Modal>
    </>
  );
};
