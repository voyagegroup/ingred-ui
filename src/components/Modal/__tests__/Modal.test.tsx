import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import Modal from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");
  return {
    ...original,
    createPortal: (node: any) => node,
  };
});

describe("Modal component testing", () => {
  afterEach(cleanup);

  test("Modal without background", () => {
    const { asFragment } = renderWithThemeProvider(
      <Modal>
        <div>hoge</div>
      </Modal>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Modal with background", () => {
    const { asFragment } = renderWithThemeProvider(
      <Modal hasBackground={true}>
        <div>hoge</div>
      </Modal>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
