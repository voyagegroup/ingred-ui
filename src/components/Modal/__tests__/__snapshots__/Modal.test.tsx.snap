// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component testing Modal with background 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cttKBd"
  >
    <div
      class="sc-dkPtRN jMueqr fade-transition-appear fade-transition-appear-active"
    >
      <div
        class="sc-gsDKAQ dLmdEA"
      />
    </div>
    <div>
      hoge
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal component testing Modal without background 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cttKBd"
  >
    <div
      class="sc-dkPtRN jMueqr fade-transition-appear fade-transition-appear-active"
    >
      <div
        class="sc-gsDKAQ dLmdEA"
      />
    </div>
    <div>
      hoge
    </div>
  </div>
</DocumentFragment>
`;
